### PR TITLE
The Janitor Strikes Back: Scrubbing (Cleaning) Skill / `remove_any` proc granularity change

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request
Adds the ~~cleaning~~ scrubbing skill. Clean up items with a mop/soap for...

- Mop: Less reagent used per swipe, Faster Mopping
- Soap: More Uses*, Faster Soaping

*RNG permitting

Also changes how `remove_any` works so it can accept numbers smaller than 1 and jumble reagent loss a bit more when dealing with lower values. I mainly tested through the janitor mop so other interactions may be "unintended", but I'd argue for the better 😏 

consequently fixes an odd portion of remove_any where you could have more removed than the amount you asked it to remove since it was expecting integers.

I decided to make the mop slightly better than the soap since the soap is more convert and smaller weight.

## Why It's Good For The Game

It's a fun skill and being able to brag about being a legendary janitor ~~and showing off your skillcape~~ is a fun story to tell.

## Changelog
:cl: Mopby
add: The Cleaning Skill
/:cl:

### CODE REVIEW

**Comment by cacogen** [comment]
hopefully, this'll help with the longstanding problem of the janitor(s) being unable to keep up with the mess

**Comment by ExcessiveUseOfCobblestone** [comment]
I have now orangified this skill, with it starting slightly slower than base but going faster than base as you progress into the deeper levels

**Comment by TheVekter** [comment]
I'm stuck at 95 mopping, what trial am I supposed to do

**Comment by ExcessiveUseOfCobblestone** [comment]
95  janitor skill is realizing you were the mess the entire time.

**Comment by HomocidlePancakes** [comment]
Janitorial ERT should have it maxed

File: code/datums/skills/cleaning.dm [hunk 1]
```diff
@@ -0,0 +1,4 @@
+/datum/skill/cleaning
+	name = "Scrubbing"
+	desc = "It’s not who I am underneath, but what I mop up that defines me."
+	modifiers = list(SKILL_SPEED_MODIFIER = list(1.15, 1.05, 1, 0.9, 0.75, 0.5, 0.36))
```
Comment (Qustinnus) [inline code/datums/skills/cleaning.dm]: im not sure if i like the use of skill_Speed_modifier if you're using it for things like efficiency of soap uses etc

**Comment by ExcessiveUseOfCobblestone** [comment]
I'm gonna deorangify the skill and see how it plays out then edit the actual numerical values.

File: code/datums/skills/cleaning.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/datums/skills/cleaning.dm]: I can leave a comment that it handles nonspeed stuff as well, I don't think there's much to gain by making an equivalent second list until I feel like they need to be different values.

**Comment by FantasticFwoosh** [comment]
Offer lizards a racial x1.75 buff to levelling up mopping to put them in their place and it will be perfect.

File: code/game/objects/items/clown_items.dm [hunk 2]
```diff
@@ -89,39 +91,47 @@
 	. = ..()
 	if(!proximity || !check_allowed_items(target))
 		return
+	var/clean_speedies = src.cleanspeed * min(user.mind.get_skill_modifier(/datum/skill/cleaning, SKILL_SPEED_MODIFIER)+0.1,1) //less scaling for soapies
```
Comment (kriskog) [inline code/game/objects/items/clown_items.dm]: Only one definition of cleanspeed here, drop the src.

File: code/game/objects/items/mop.dm [hunk 3]
```diff
@@ -50,10 +55,10 @@
 
 	if(T)
 		user.visible_message("<span class='notice'>[user] begins to clean \the [T] with [src].</span>", "<span class='notice'>You begin to clean \the [T] with [src]...</span>")
-
-		if(do_after(user, src.mopspeed, target = T))
+		var/clean_speedies = user.mind.get_skill_modifier(/datum/skill/cleaning, SKILL_SPEED_MODIFIER)
+		if(do_after(user, src.mopspeed*clean_speedies, target = T))
```
Comment (kriskog) [inline code/game/objects/items/mop.dm]: This src too

**Review by kriskog — CHANGES_REQUESTED** [review]
Nitpicky and doesnt change behaviour, but yknow.

File: code/game/objects/items/mop.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/mop.dm]:
```suggestion
		if(do_after(user, mopspeed*clean_speedies, target = T))
```

File: code/game/objects/items/clown_items.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/clown_items.dm]:
```suggestion
	var/clean_speedies = cleanspeed * min(user.mind.get_skill_modifier(/datum/skill/cleaning, SKILL_SPEED_MODIFIER)+0.1,1) //less scaling for soapies
```

**Review by kriskog — APPROVED** [review]

**Pull request merged**